src/core/dbus.c

```diff
@@ -522,8 +477,139 @@ static int bus_unit_enumerate(sd_bus *bus, const char *path, void *userdata, cha
         return k;
 }
 
+static const BusObjectImplementation unit_object = {
+        "/org/freedesktop/systemd1/unit",
+        "org.freedesktop.systemd1.Unit",
+        .vtables = VTABLE_LIST( { bus_unit_vtable,        bus_unit_find } ),
+        .node_enumerator = bus_unit_enumerate,
+        .children = CHILD_LIST(),
+};
```

Comment:
what's the rationale for having these structues here instead of the per-unit-type files? i.e. why export the vtable and the find routine instead of the structure?

Reply:
I started the other way around, and it seemed more messy, and in the end I reworked it to be this way. The vtable needs to be exported anyway for the code that implements `--dump-properties`.

Reply:
Oh, I now remember: those functions use helpers which are defined in dbus.c. To move them to the other files would require making the helpers non-static, which in the end results in a lot added lines.